src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -584,25 +555,37 @@ CHIPEventOpenCL *CHIPBackendOpenCL::createCHIPEvent(CHIPContext *ChipCtx,
 }
 
 void CHIPEventOpenCL::recordStream(CHIPQueue *ChipQueue) {
+  LOCK(Backend->EventsMtx); // trackImpl CHIPBackend::Events
+  LOCK(EventMtx);           // changing this event's fields
   logTrace("CHIPEvent::recordStream()");
-  auto MarkerEvent = ChipQueue->enqueueMarker();
-  this->takeOver(MarkerEvent);
 
-  this->EventStatus_ = EVENT_STATUS_RECORDING;
+  CHIPEventOpenCL *Marker = (CHIPEventOpenCL *)ChipQueue->enqueueMarkerImpl();
+  // see operator=() on cl::Event
+  // should automatically release ClEvent if it already contains valid handle
+  ClEvent = Marker->ClEvent;
+  Msg = "recordStreamMarker";
+  EventStatus_ = EVENT_STATUS_RECORDING;
+  delete Marker;
+
+  ChipQueue->updateLastEvent(this);
+  // can't use this->track() because it calls locks
+  if (!TrackCalled_) {
+    Backend->Events.push_back(this);
+    TrackCalled_ = true;
+  }
+
   return;
 }
 
-void CHIPEventOpenCL::takeOver(CHIPEvent *OtherIn) {
-  logTrace("CHIPEventOpenCL::takeOver");
-  decreaseRefCount("takeOver");
-  {
-    auto *Other = (CHIPEventOpenCL *)OtherIn;
-    LOCK(EventMtx); // CHIPEvent::Refc_
-    this->ClEvent = Other->ClEvent;
-    this->Refc_ = Other->Refc_;
-    this->Msg = Other->Msg;
+size_t CHIPEventOpenCL::getCHIPRefc() {
+  int Err = CL_SUCCESS;
+  size_t RefC = ClEvent.getInfo<CL_EVENT_REFERENCE_COUNT>(&Err);
+  if (Err != CL_SUCCESS) {
+    logError("failed to get Reference count from OpenCL event");
+    return 0;
+  } else {
+    return RefC;
   }
```

Comment:
Style nit (no else after return):
```suggestion
  if (Err != CL_SUCCESS) {
    logError("failed to get Reference count from OpenCL event");
    return 0;
  }
  return RefC;
```